### PR TITLE
feat: Launch apps as platforms and inter-platform interop

### PR DESCRIPTION
**Pull request by TobyLi96**
- Launch FX/Credit/Limit Checker/Reactive Analytics as platforms (enables snapshot)
- Add inter-platform interop to keep interop between FX and Reactive Analytics/Limit Checker working

### CODE REVIEW

File: src/client/OpenFin/Window/WindowControls.tsx
Comment (SHolleworth) [inline src/client/OpenFin/Window/WindowControls.tsx:1]: isReactiveTraderPlatformPrimary can be taken out of the imports

**Review by SHolleworth — COMMENTED** [review]
This looks fine to me but I don't understand this area very well, I would hold off and let Alan look through it when he gets back

File: src/client/OpenFin/provider.ts [hunk 1]
```diff
@@ -1,29 +1,217 @@
...
+    public limitCheckerUuid: string
+
+    public analyticsClients: ExternalClientMap
+
+    public limitCheckerClients: ExternalClientMap
+
+    constructor() {
+      super()
+      this.analyticsUuid = `reactive-analytics-${ENVIRONMENT}`
+      this.limitCheckerUuid = `reactive-trader-limit-checker` // TODO make env specific
```
Comment (TobyLi96) [inline src/client/OpenFin/provider.ts:30]: I'm not sure that the limit checker UUID is env specific?

Comment (algreasley) [reply src/client/OpenFin/provider.ts:30]: it should be

Comment (TobyLi96) [reply src/client/OpenFin/provider.ts]: Done

File: src/client/OpenFin/provider.ts [hunk 2]
```diff
@@ -1,29 +1,217 @@
...
+      externalUuid: string,
+      externalClients: ExternalClientMap,
+    ) {
+      // Create a InteropClient instance by connecting to the external broker
+      const externalInteropClient = fin.Interop.connectSync(externalUuid, {})
+      const externalContextGroup =
+        await externalInteropClient.getContextGroups()
+
+      // Create a InteropClient instance by connecting to the current platform broker
+      const fxInteropClient = fin.Interop.connectSync(fin.me.uuid, {})
```
Comment (algreasley) [inline src/client/OpenFin/provider.ts:75]: minor: probably doesn't matter for this pass, but can we remove the fx-specificity (or is this not generic enough for Credit yet?)

File: src/client/OpenFin/Window/WindowControls.tsx [hunk 3]
```diff
@@ -39,14 +35,11 @@ export const WindowControls = ({ close, minimize, maximize, popIn }: Props) => {
   }, [])
 
   async function wrappedClose() {
+    // ONLY if RT/LC main win is primary win of platform, close all platform windows
```
[the remark this answers is not in the document]
Comment (TobyLi96) [reply src/client/OpenFin/Window/WindowControls.tsx]: Moved back

File: src/client/OpenFin/Window/WindowControls.tsx [hunk 4]
```diff
@@ -39,14 +39,11 @@ export const WindowControls = ({ close, minimize, maximize, popIn }: Props) => {
...
     if (inReactiveTraderMainWindow()) {
-      // ONLY if RT main win is primary win of platform, close all platform windows
-      if (isReactiveTraderPlatformPrimary()) {
-        fin.Platform.getCurrentSync().quit()
-        return
-      }
-      // otherwise close all windows related to the main RT (FX OR Credit)
-      await closeOtherWindows()
+      const windows = await fin.Application.getCurrentSync().getChildWindows()
```
[the remark this answers is not in the document]
Comment (TobyLi96) [reply src/client/OpenFin/Window/WindowControls.tsx]: Moved to `utils/windows.ts` and added reference to OF issue

File: src/client/OpenFin/provider.ts (hunk 2 above)
Comment (TobyLi96) [reply src/client/OpenFin/provider.ts]: Renamed

File: src/client/OpenFin/provider.ts [hunk 5]
```diff
@@ -1,29 +1,217 @@
...
+      }
+    }
+
+    /**
+     * @override @method setContext
+     * @param payload object that is passed in when set context is called from an OpenFin entity using the interop api.
+     * @param clientIdentity object containing the clientIdentity of the sender.
+     * @example // please refer to the working examples code panel in this projects interface.
+     */
+    public async setContext(
```
[the remark this answers is not in the document]
Comment (TobyLi96) [reply src/client/OpenFin/provider.ts]: Removed `public` from `setContext`

File: src/client/OpenFin/provider.ts [hunk 6]
```diff
@@ -1,29 +1,217 @@
...
+>
+
+export interface ExternalContext extends OpenFin.Context {
+  _clientInfo?: {
+    uuid: string
+  }
+}
+
+function interopOverride(
+  InteropBroker: OpenFin.Constructor<OpenFin.InteropBroker>,
```
[the remark this answers is not in the document]
Comment (TobyLi96) [reply src/client/OpenFin/provider.ts:16]: I'm not quite sure what to comment here. I assumed it was just the standard boiler plate code

[the remark this answers is not in the document]
Comment (algreasley) [reply src/client/OpenFin/provider.ts:16]: it is standard boilerplate, per OpenFin, just odd-looking code .. might wrong-foot the unsuspecting ..
dw, leave it with me .. I'll be in the FDC3 area later

**Review by algreasley — APPROVED** [review]
Nice work Toby .. very pleased to get the Platforms back and working properly with interop
(pity about the Platform.quit 😞)

**Pull request merged**